### PR TITLE
Issue samvirk#13/exchange rate provider

### PR DESCRIPTION
**Pull request by markiyanshvets**
Background service will update NOK currency rate as a base every 2 hours and upsert it into the database. If the requested currency cannot be converted due to absence in the database, it will request to get a new base currency and write it to DB. In the test issue, there is an endpoint to use currency exchange. 
Each base currency type has a validation per request (2 hours maximum lifetime)
Allowable currencies depend on Enums.Currencies

### CODE REVIEW

File: Api/BccPay.Core.Infrastructure/Helpers/CurrencyService.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+            if (lastCurrencyRateUpdate is not null)
+            {
+                lastCurrencyRateUpdate.Update(rates, TimeStampConverter.UnixTimeStampToDateTime(fixerApiResult.Timestamp));
+            }
+
+            await _documentSession.SaveChangesAsync(cancellationToken);
+        }
+
+        private async Task<(decimal, bool)> GetExhangeRateByCurrency(Currencies formCurrency, Currencies toCurrency)
```
Comment (tkristiansen) [inline Api/BccPay.Core.Infrastructure/Helpers/CurrencyService.cs:99]: from, not form :-)

File: Api/BccPay.Core.Infrastructure/Helpers/CurrencyService.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            decimal exchangeResultNetto = 0;
+            if (fromOpposite)
+                exchangeResultNetto = Decimal.Divide(amount, currencyRate);
+            if (!fromOpposite)
+                exchangeResultNetto = Decimal.Multiply(amount, currencyRate);
+
+
+            if (tax is not 0)
+            {
+                tax += Decimal.Multiply(exchangeResultNetto, tax);
```
Comment (tkristiansen) [inline Api/BccPay.Core.Infrastructure/Helpers/CurrencyService.cs:51]: I don't understand what you are doing here... += multiplied value?

**Review by tkristiansen — CHANGES_REQUESTED** [review]
* I suggest to use a deterministic document id for the currency rate (i.e. curencyRate/[base-currency]). That way, you don't have to index / query for the document, but can just load it directly.
* The API should not include the `tax` field. This should be hidden configuration.
* We should also not call it tax, since that is a bit confusing. ExchangeRateMarkup seems better.

**Comment by tkristiansen** [comment]
@markiyanshvets, another comment regarding this:

>The API should not include the tax field. This should be hidden configuration

I think the exchange rate markup should practically be part of the payment method config. Maybe you can take the payment method key as argument in the API call instead of the markup itself. Then you can always use this API to get the correct amount using a certain payment method?

**Comment by markiyanshvets** [comment]
> * I suggest to use a deterministic document id for the currency rate (i.e. curencyRate/[base-currency]). That way, you don't have to index / query for the document, but can just load it directly.
> * The API should not include the `tax` field. This should be hidden configuration.
> * We should also not call it tax, since that is a bit confusing. ExchangeRateMarkup seems better.

Partially changed. 
I think, we will make this converter internal and will use it inside payment provider concrete implementations. Currently, it is in Core.API to test it manually.

**Review by tkristiansen — APPROVED** [review]

**Pull request merged**